source/agora/node/Ledger.d

```diff
@@ -230,6 +234,84 @@ public class Ledger
         else
             return null;
     }
+
+    /***************************************************************************
+
+        Write the block on the ledgerFile.
+
+        Params:
+            T = Type of struct to writeFile
+            record = Instance of `T` to serialize
+            dg  = Serialization delegate, when this struct is a nested struct
+
+    ***************************************************************************/
+
+    public void writeBlockFile (T) (auto ref T record) @safe
```

Comment:
If this only writes blocks, why is it templated? You could just write `( Block record)`

Reply:
apply :)

Reply:
public void writeBlockFile (ref Block block) @safe